frontend/src/metabase/query_builder/components/view/View/NotebookContainer/NotebookContainer.tsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from "react";
+
+import Notebook from "metabase/query_builder/components/notebook/Notebook";
+import { Box } from "metabase/ui";
+
+// There must exist some transition time, no matter how short,
+// because we need to trigger the 'onTransitionEnd' in the component
+const delayBeforeNotRenderingNotebook = 10;
+
+interface NotebookContainerProps {
+  isNotebookContainerOpen: boolean;
+}
+
+export const NotebookContainer = ({
+  isNotebookContainerOpen,
+  ...props
+}: NotebookContainerProps) => {
+  const [shouldShowNotebook, setShouldShowNotebook] = useState(
+    isNotebookContainerOpen,
+  );
+
+  useEffect(() => {
+    isNotebookContainerOpen && setShouldShowNotebook(isNotebookContainerOpen);
+  }, [isNotebookContainerOpen]);
+
+  const handleTransitionEnd: React.TransitionEventHandler<HTMLDivElement> = (
+    event,
+  ): void => {
+    if (event.propertyName === "opacity" && !isNotebookContainerOpen) {
+      setShouldShowNotebook(false);
+    }
+  };
+
+  const transformStyle = isNotebookContainerOpen
+    ? "translateY(0)"
+    : "translateY(-100%)";
+
+  return (
+    <Box
+      bg="white"
+      pos="absolute"
+      top={0}
+      bottom={0}
+      left={0}
+      right={0}
```

Comment:
A shorter way of doing the same:

```suggestion
      inset={0}
```

Reply:
https://github.com/metabase/metabase/pull/40316/commits/ede302543e87bd1a56d1d15d38fce11a67d6b23e